src/abstract_block.jl

```diff
@@ -5,16 +5,19 @@ using YaoBase, YaoArrayRegister, SimpleTraits
 
 Apply a block (of quantum circuit) to a quantum register.
 """
+function apply!(r::AbstractRegister, @nospecialize(b::AbstractBlock))
```

Comment:
why do we need a `nospecialize` here?

Reply:
I do not think it will make much difference here. Intuitively there are many types feeding into this function, we do not need to recompile each of them. Which is better? I am not sure.